src/module/actor/base.ts

```diff
@@ -710,8 +713,12 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
 
         const initiativeRoll = Math.trunc(participant.initiative);
         rollOptionsAll[`self:participant:initiative:roll:${initiativeRoll}`] = true;
-        const rank = [...participants].reverse().indexOf(participant) + 1;
+        const rank = participants.indexOf(participant) + 1;
         rollOptionsAll[`self:participant:initiative:rank:${rank}`] = true;
+        if (participant.flags.pf2e.initiativeStatistic) {
+            const stat = participant.flags.pf2e.initiativeStatistic;
```

Comment:
Is it reasonable to pull this out of the if block so it can be used in the if test itself?

Reply:
I don't follow

Reply:
`const stat = participant.flags.pf2e.initiativeStatistic;
if (stat) {`

Reply:
There's no sense in setting a roll option of `"self:participant:initiative:stat:null": false`.

Reply:
`if (stat) {`
Should address that.
This is more about the cost of accessing the properties twice, which is probably negligible anyway.

Reply:
Its a very minor change to strip out that doesn't save any lines. It'd be a better idea if this was javascript but in typescript its not a big deal.